### PR TITLE
Finish

### PR DESCRIPTION
**Pull request by Ultramarin**
Solution for task stateful-object

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -6,6 +6,25 @@
...
+        delete state[key];
+      });
+    }
+
+    if (action['type'] === 'clear') {
+      Object.keys(state).forEach(key => {
+        delete state[key];
+      });
+    }
+  }
```
Comment (VitaliyBondarenko1982) [inline src/transformState.js:27]: Better to use switch...case

File: src/transformState.js [hunk 2]
```diff
@@ -6,6 +6,25 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const action of actions) {
+    if (action['type'] === 'addProperties') {
+      Object.keys(action['extraData']).forEach(key => {
+        state[key] = action['extraData'][key];
+      });
```
Comment (VitaliyBondarenko1982) [inline src/transformState.js:13]:
```suggestion
    Object.assign(state, extraData);
```

File: src/transformState.js [hunk 3]
```diff
@@ -6,6 +6,25 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const action of actions) {
```
Comment (VitaliyBondarenko1982) [inline src/transformState.js:9]:
```suggestion
  for (const action of actions) {
  const { type, extraData, keysToRemove } = action
```

File: src/transformState.js [hunk 4]
```diff
@@ -6,6 +6,25 @@
...
+    if (action['type'] === 'addProperties') {
+      Object.keys(action['extraData']).forEach(key => {
+        state[key] = action['extraData'][key];
+      });
+    }
+
+    if (action['type'] === 'removeProperties') {
+      action['keysToRemove'].forEach(key => {
```
Comment (VitaliyBondarenko1982) [inline src/transformState.js:17]:
```suggestion
      keysToRemove.forEach(key => {
```

File: src/transformState.js [hunk 5]
```diff
@@ -6,6 +6,25 @@
...
+        delete state[key];
+      });
+    }
+
+    if (action['type'] === 'clear') {
+      Object.keys(state).forEach(key => {
+        delete state[key];
+      });
```
Comment (VitaliyBondarenko1982) [inline src/transformState.js:25]:
```suggestion
       for (const key in state) {
          delete state[key];
        }
```

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Good job, but needs some refactor

File: src/transformState.js (hunk 1 above)
Comment (Ultramarin) [reply src/transformState.js]: I change if to wsitch

File: src/transformState.js (hunk 3 above)
Comment (Ultramarin) [reply src/transformState.js:9]: Change

File: src/transformState.js (hunk 5 above)
Comment (Ultramarin) [reply src/transformState.js]: Change

File: src/transformState.js (hunk 4 above)
Comment (Ultramarin) [reply src/transformState.js]: I change

File: src/transformState.js (hunk 2 above)
Comment (Ultramarin) [reply src/transformState.js]: I change change

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]
image

**Comment by Ultramarin** [comment]
> image

I fix this

**Pull request closed without merging**